github/git_commits.go

```diff
@@ -133,3 +150,51 @@ func (s *GitService) CreateCommit(ctx context.Context, owner string, repo string
 
 	return c, resp, nil
 }
+
+func createSignature(signingKey *openpgp.Entity, commit *createCommit) (string, error) {
+	if signingKey == nil || commit == nil {
+		return "", errors.New("createSignature: invalid parameters")
+	}
+
+	message, err := createSignatureMessage(commit)
+	if err != nil {
+		return "", err
+	}
+
+	writer := new(bytes.Buffer)
+	reader := bytes.NewReader([]byte(message))
+	if err := openpgp.ArmoredDetachSign(writer, signingKey, reader, nil); err != nil {
+		return "", err
+	}
+
+	return writer.String(), nil
+}
+
+func createSignatureMessage(commit *createCommit) (string, error) {
+	if commit == nil || commit.Message == nil || commit.Author == nil {
+		return "", errors.New("createSignatureMessage: invalid parameters")
+	}
+
+	var message []string
+
+	if commit.Tree != nil {
+		message = append(message, fmt.Sprintf("tree %s", *commit.Tree))
+	}
+
+	for _, parent := range commit.Parents {
+		message = append(message, fmt.Sprintf("parent %s", parent))
+	}
+
+	message = append(message, fmt.Sprintf("author %s <%s> %d %s", commit.Author.GetName(), commit.Author.GetEmail(), commit.Author.GetDate().Unix(), commit.Author.GetDate().Format("-0700")))
+
+	committer := commit.Committer
+	if committer == nil {
+		committer = commit.Author
+	}
+
+	// There needs to be a double newline after committer
+	message = append(message, fmt.Sprintf("committer %s <%s> %d %s\n", committer.GetName(), committer.GetEmail(), committer.GetDate().Unix(), committer.GetDate().Format("-0700")))
```

Comment:
On this line you can remove the `\n`.

Reply:
Yea I was debating what to do with this one. We need 2 new lines after the committer for this the be valid. It was between this or doing an empty string 
```
	message = append(message, fmt.Sprintf(""))
```
Though not sure if that is better or not.

Reply:
Oh, sorry... yes, if you want a blank line definitely keep the `\n` here.  No, never use `fmt.Sprintf("")`... that would just be a `""` string literal... but in this case, please just keep the `\n`.